web/e2e/driver/driver.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+
+import { type WebDriver, Builder } from 'selenium-webdriver'
+import fetch from 'node-fetch'
+import childProcess from 'child_process'
+import serverConfigs from '../config/configs'
+import { clone } from 'lodash'
+import type { EndToEndConfigType } from '../config/configs'
+
+const BROWSERSTACK_EXHAUSTED_MESSAGE = 'All parallel tests are currently in use, including the queued tests. ' +
+  'Please wait to finish or upgrade your plan to add more sessions.'
+const IMPLICIT_WAIT_TIMEOUT = 80000
+const STARTUP_DELAY = 10000
+
+export const timer = (ms: number) => new Promise<void>(resolve => setTimeout(resolve, ms))
+
+const getConfig = (): EndToEndConfigType | null => {
+  const configName: ?string = process.env.E2E_CONFIG
+
+  if (!configName) {
+    console.error('E2E_CONFIG name is not set!')
+    return null
+  }
+
+  const config = serverConfigs[configName.toLowerCase()]
+
+  if (!config) {
+    console.error(`Server config ${configName} not found!`)
+    return null
+  }
+
+  return config
+}
+
+const initDriver = async (config: EndToEndConfigType, desiredCaps): Promise<WebDriver> => {
```

Comment:
Why no type for desiredCaps?

Reply:
They can be arbitrary objects. But we could define a type for them. If so then this change should also be done in native

Reply:
for now we do not have any use case for desired caps in web anyway. I just added them as native also has them. Should I remove them for now?

Reply:
Your call :) I think leaving it makes sense for consistency.